tests/ipfwd/test_nhop_count.py

```diff
@@ -0,0 +1,178 @@
+            for ip_mac in self.arp_list:
+                cmd = arp_cmd.format(self.asic.ns_arg, ip_mac.ip, ip_mac.mac)
+                fn.write(cmd + "\n")
+
+        self.duthost.copy(src=self.filename, dest=self.filename, mode=0755)
+        result = self.duthost.shell(self.filename)
+        pytest_assert(
+            result["rc"] == 0,
+            "arp add failed on duthost:{}".format(self.filename)
+        )
+        self.duthost.shell("rm {}".format(self.filename))
+
+    def arps_del(self):
+        """
+        Create a file with static arp del commands, copy file
+        to DUT and run it from DUT
+        """
+        arp_cmd = "sudo {} arp -d {}"
+        with open(self.filename, "w") as fn:
+            for ip_mac in self.arp_list:
+                cmd = arp_cmd.format(self.asic.ns_arg, ip_mac.ip)
+                fn.write(cmd + "\n")
+
+        self.duthost.copy(src=self.filename, dest=self.filename, mode=0755)
+        try:
+            self.duthost.shell(self.filename)
+            self.duthost.shell("rm {}".format(self.filename))
+        except:
+            pass
+
+        os.remove(self.filename)
+
+    def add_ip_route(self, ip="192.168.5.0/24"):
+        """
+        Add IP route with ECMP paths via 'vtysh' command
+        """
+        self.arps_add()
+        self.ip = ip
+        vty_cmd = "-c 'config term'"
+
+        for ip_mac in self.arp_list:
+            vty_cmd += " -c 'ip route {} {}'".format(self.ip, ip_mac.ip)
+
+        vty_cmd += " -c 'exit'"
+
+        result = self.asic.run_vtysh(vty_cmd)
+        pytest_assert(
+            result["rc"] == 0, "Route add failed:{}".format(vty_cmd)
+        )
+
+    def clean_up(self):
+        # delete ip route
+        if self.ip:
+            vty_cmd = "-c 'config term'"
+
+            for ip_mac in self.arp_list:
+                vty_cmd += " -c 'no ip route {} {}'".format(self.ip, ip_mac.ip)
+
+            vty_cmd += "-c 'exit'"
+            logging.info("ROUTE DEL {}".format(vty_cmd))
+
+            try:
+                self.asic.run_vtysh(vty_cmd)
+            except:
+                pass
+
+        # delete static ARPs
+        self.arps_del()
+
+        # del IP address from the eth interface
+        ip_iface = "ip address del {} dev {}".format(self.if_addr, self.iface)
+        logging.info("IF ADDR DEL {}".format(ip_iface))
+        try:
+            self.asic.command(ip_iface)
+        except:
+            pass
+
+def test_nhop(request, duthost):
+
+    default_max_nhop_paths = 32
+    nhop_group_limit = 512
+
+    asic = duthost.asic_instance()
+
+    # find out MAX NHOP group count supported on the platform
+    result = asic.run_redis_cmd(
+        argv = ["redis-cli", "-n", 6, "HGETALL", "SWITCH_CAPABILITY|switch"]
+    )
+    it = iter(result)
+    switch_capability = dict(zip(it, it))
+    max_nhop = switch_capability.get("MAX_NEXTHOP_GROUP_COUNT")
+    max_nhop = default_max_nhop_paths if max_nhop == None else int(max_nhop)
```

Comment:
if the max_nexthop_group_count is set to 256, then the max_paths will also be 256 ?
The test might be exceed the hardware limit ?

Reply:
I think there's confusion in the terminology. "NEXTHOP_GROUP_COUNT" is actually the number of paths in the nexthop. I'm creating only one nexthop. "max_nop" is the number of paths. 

I'll change it to "max_nhop_paths" if it helps.

Reply:
Thanks, I got confused `NEXTHOP_GROUP_COUNT`.  changing the variable to max_nhop_paths will help.